src/contracts/atlas/GasAccounting.sol

```diff
@@ -127,8 +127,10 @@ abstract contract GasAccounting is SafetyLocks {
     /// @return valid A boolean indicating whether the borrowing operation was successful.
     function _borrow(uint256 amount) internal returns (bool valid) {
         if (amount == 0) return true;
+
         uint256 _withdrawals = withdrawals + amount;
-        if (address(this).balance < claims + _withdrawals) return false;
```

Comment:
@jj1980a @thogard785 note, I had to also remove `claims` from this calculation to make it possible for a solver to borrow the full Atlas balance. But this should be okay for the same reasons discussed at the start of this PR.

Reply:
FYI there is a very similar but different change for this calculation in this fix https://github.com/FastLane-Labs/atlas/pull/234

Reply:
Yeah, that's a very relevant issue, good catch.

So I suppose the combination of that isssue + your PR, and this issue + PR, is that the check ends up as:
`if (address(this).balance < amount) return false;`

- `withdrawals` is excluded because that amount is already not part of contract's balance (see: https://github.com/spearbit-audits/review-fastlane/issues/136)
- `claims` is also excluded because the loan _must_ be repaid, otherwise `_settle()` will revert and the borrowed value will not have left the contract effectively. This is made more secure by the change where `borrow()` will be disabled from post Solver phase to the end of the metacall (see: https://github.com/spearbit-audits/review-fastlane/issues/18)